### PR TITLE
Feature/ms/t4.3 join community

### LINKED ISSUE #7: T5.3: Develop PUT endpoint that removes a user from a course community.

### LINKED ISSUE #8: T4.3: Develop PUT endpoint that registers a user to a course community.

### LINKED ISSUE #10: T2.3: Create GET endpoint that returns list of joined communities of a user.

### LINKED ISSUE #37: T5.4: Write tests for the components

### LINKED ISSUE #38: T4.4: Write tests for the components

### LINKED ISSUE #39: T2.4: Write tests for the components

### PR DESCRIPTION
**Pull request by msalze**
Implement joining/ leaving and retrieving information about user and their communities.
Test the implemented functionality.

Closes #10 .
Closes #39 .
Closes #8 .
Closes #38 .
Closes #7 .
Closes #37 .

### CODE REVIEW

File: src/test/java/com/agileavengers/icuconnectbackend/service/ICommunityServiceTest.java [hunk 1]
```diff
@@ -41,12 +39,13 @@
...
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 
 import static org.mockito.Mockito.*;
 
 @ExtendWith({SpringExtension.class, MockitoExtension.class})
 @SpringBootTest()
-@ComponentScan(basePackageClasses = IcuConnectBackendApplication.class)
+//@ComponentScan(basePackageClasses = IcuConnectBackendApplication.class)
```
Comment (solodezaldivar) [inline src/test/java/com/agileavengers/icuconnectbackend/service/ICommunityServiceTest.java:48]: Can we remove this?

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+@SpringBootTest()
+class UserServiceTest {
+
+    IUserService userService;
+
+    @Mock
+    CommunityRepository communityRepository;
+    //    @Mock
```
Comment (solodezaldivar) [inline src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:46]: Can we remove this?

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+//    InstructorRepository instructorRepository;
```
Comment (solodezaldivar) [inline src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:47]: And this?

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+                user.getPassword(), new ArrayList<>()));
+        SecurityContext securityContext = mock(SecurityContext.class);
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+        SecurityContextHolder.setContext(securityContext);
+    }
+
+    @BeforeEach
+    void setup() {
+        communityRepository = mock(CommunityRepository.class);
+//        instructorRepository = mock(InstructorRepository.class);
```
Comment (solodezaldivar) [inline src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:68]: Here as well

File: src/test/java/com/agileavengers/icuconnectbackend/service/JwtUserDetailsServiceTest.java [hunk 5]
```diff
@@ -1,39 +1,34 @@
...
+import static org.junit.gen5.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
 
 @SpringBootTest()
 @ExtendWith({ SpringExtension.class, MockitoExtension.class })
-@ComponentScan(basePackageClasses = IcuConnectBackendApplication.class)
+//@ComponentScan(basePackageClasses = IcuConnectBackendApplication.class)
```
Comment (solodezaldivar) [inline src/test/java/com/agileavengers/icuconnectbackend/service/JwtUserDetailsServiceTest.java:31]: Can we remove this?

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+        communityRepository = mock(CommunityRepository.class);
+        userRepository = mock(UserRepository.class);
+        ratingRepository = mock(RatingRepository.class);
+        MappingService mappingService = new MappingService(userRepository, ratingRepository);
+        CommunityMapper communityMapper = Mappers.getMapper(CommunityMapper.class);
+        communityMapper.setMappingService(mappingService);
+        RatingMapper ratingMapper = new RatingMapperImpl(communityMapper);
+        ratingMapper.setMappingService(mappingService);
+//        InstructorMapper instructorMapper = Mappers.getMapper(InstructorMapper.class);
```
Comment (solodezaldivar) [inline src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:77]: Can we remove this?

**Review by solodezaldivar — APPROVED** [review]
👍

File: src/test/java/com/agileavengers/icuconnectbackend/service/ICommunityServiceTest.java (hunk 1 above)
Comment (msalze) [reply src/test/java/com/agileavengers/icuconnectbackend/service/ICommunityServiceTest.java:48]: Done

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java (hunk 2 above)
Comment (msalze) [reply src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:46]: Done

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java (hunk 3 above)
Comment (msalze) [reply src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:47]: Done

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java (hunk 4 above)
Comment (msalze) [reply src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:68]: Done

File: src/test/java/com/agileavengers/icuconnectbackend/service/JwtUserDetailsServiceTest.java (hunk 5 above)
Comment (msalze) [reply src/test/java/com/agileavengers/icuconnectbackend/service/JwtUserDetailsServiceTest.java:31]: Done

File: src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java (hunk 6 above)
Comment (msalze) [reply src/test/java/com/agileavengers/icuconnectbackend/service/implementation/UserServiceTest.java:77]: Done

**Review by solodezaldivar — APPROVED** [review]

**Pull request merged**